### PR TITLE
Cascade control (simple use-case for chaining controllers)

### PR DESCRIPTION
**Pull request by destogl**

### CODE REVIEW

**Comment by bijoua29** [comment]
@destogl Looks good. It will be good to get a initial proof of concept for the simple use-case and then expand further from there. Some comments:

- I like the explicitness of a ChainableController type although adding chainable interface in ControllerInterface would work too. I like however knowing/setting explicitly if a controller is chainable. In the future, the requirements for a chainable controller may expand and having it in a separate class seems better.
- If you include a controller in the chained_controlllers configuration, would it default to chained_mode being true?
- when deactivating a controller, would you reset the chained_mode to false?
- would controller manager expose services to enable/disable chained mode for a controller or would that be a parameter when activating a controller? Related: would you need to deactivate a controller first to change its 'chained_mode' state

**Comment by destogl** [comment]
> * If you include a controller in the chained_controlllers configuration, would it default to chained_mode being true?
> 
> * when deactivating a controller, would you reset the chained_mode to false?
> 
> * would controller manager expose services to enable/disable chained mode for a controller or would that be a parameter when activating a controller? Related: would you need to deactivate a controller first to change its 'chained_mode' state

The idea is to let controller manager doing internally management of chained controller depending on how they are currently used. The idea of this flag, i.e., method, is that a controller knows when it is used in chained mode and disables its subscribers. I hope this would increase clarity when using controllers in chained mode, since they can get input only from preceding controllers and not additionally from subscribers.

Although I am not 100% sure that this approach doesn't cause some limitations in specific scenarios. I am happy to adjust this after having the first implementation working.

File: design_drafts/cascade_control.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+      - parallel_group1:
+        - position_tracking
+
+      - parallel_group2:
+        - diff_drive_controller
+
+      - parallel_group3:
+        - pid_left_wheel
+        - pid_right_wheel
```
Comment (destogl) [inline design_drafts/cascade_control.md:97]:
```suggestion
controller_manager:
  ros__parameters:
    chained_controllers:

      - parallel_group_1:
        - controller1_1
        - controller1_2

      - parallel_group_2:
        - controller2_1

      - parallel_group_3:
        - controller3_1
        - controller3_2
        - controller3_3

      ...

      - parallel_group_N:
        - controllerN_1
        - ...
        - controllerN_M
```

#### Example
```
controller_manager:
  ros__parameters:
    chained_controllers:

      - parallel_group_1:
        - position_tracking_controller

      - parallel_group_2:
        - diff_drive_controller

      - parallel_group_3:
        - pid_left_wheel_controller
        - pid_right_wheel_controller
```

**Review by livanov93 — APPROVED** [review]
This looks good to me like a roadmap doc. Still, more will be known when a simple demo is created.

**Pull request merged**